### ISSUE TITLE
x11docker dependencies

### ISSUE THREAD

**Issue by domichel** [mviereck/x11docker#536]
I an making a gentoo ebuild for x11docker. For all optional features, gentoo have a policy of USE flags. With them, the users can choose if they want (USE="myflag") or not (USE="-myflag") to install the myflag support. Which imply it will be a lot of them into that ebuild, and they will be used to trigger the installation of the needed/wanted dependencies. I also want to be able to make my own docker images.

When I read

> This recommended base allows advanced usage of x11docker and an improved X security setup. Some of them are probably already installed on your system.
> 
> Desired commands: nxagent xinit xauth xclip xhost xrandr xdpyinfo catatonit jq

does that mean that I will need all of these commands? And does that mean I need them in the host?

And with

> Recommended additional base for --gpu
> Desired commands: weston Xwayland xdotool xpra

Does that mean that I need both weston and Xwayland with wayland, or both of xdotool and xpra with xorg? And does I need them only in the images, or also in the host?

**Comment by domichel** [comment]
Making an image, I begin to understand how the whole thing is working. So  we can close that bug.

**Comment by mviereck** [comment]
Great that you found out yourself. :-)
Is your gentoo ebuild intended to be for public use / in some official repository?

If you have further questions, please ask. Unfortunately I am only rarely at this project nowadays.

**Comment by domichel** [comment]
Yes, I will publish it into an overlay. When done I will also make a bug report on gentoo asking for a new package.

I have 2 public docker images on the way, one based on Debian 10 for the dkbuilder (toolkit for generation and building of guitarix and lv2 plugins from electronic schematics), the other one for x11docker based on Debian unstable and running fvwm-crystal from git.

**Comment by mviereck** [comment]
> Yes, I will publish it into an overlay. When done I will also make a bug report on gentoo asking for a new package.

Great, thank you!

You might find these hints useful:
https://github.com/mviereck/x11docker/wiki/Packaging-x11docker

**Comment by domichel** [comment]
I  saw than on gentoo, the X tools dependencies need x11-apps/setxkbmap too.

**Comment by domichel** [comment]
I have done a gentoo bug for a new package: https://bugs.gentoo.org/941144

**Comment by domichel** [comment]
I published it here: https://github.com/domichel/GenCool
It's not an official gentoo overlay, but it is referenced here, which is an official gentoo overlay: https://github.com/domichel/proaudio-gentoo
